Converted tests to Test.jl

Frosco opened the pull request:
So, here goes the first real tricky one...

After the first round of edits, a couple of tests were failing, so I gave it another go of looking into it. The success return value issue seems fine, but the rest...
Two tests were fixed by increasing the allowed difference between FEM and analytical tests to 95%, which is... a lot? It was motivated by implementation differences.
Also, some tests had boundary condition issues and the model chose a different approach to make it work, don't know if that's allowed :P 

So for an outsider, all this seems fishy at least. You can check the "fix" commit so see what was changed.

Although all the other tests are passing, it would probably be beneficial to check that they are equivalent to the test scripts that you wrote, so that we're actually testing the right thing. I'm a bit out of my depth here :see_no_evil: 

The tests can't be run with the standard `Pkg.test()` functionality yet, because we don't have a main module file, but that comes later, when the refactoring starts in earnest. For now, you have to run the files manually.

Frosco committed:
Added performance tests

To verify our refactoring is not going to impact performance.

Frosco committed:
Converted test scripts to Test.jl framework

Frosco committed:
Fixed tests

jgrashorn commented:
The differences might stem from the fact that somehow the natural frequencies ($\lambda$) are converted again into Hz:

`test/integration/test_beam_dynamic.jl`, line 289

```julia
frequencies = sqrt.(λ[:, 1]) / (2*π)  # Convert to Hz
```

 This is already done inside `decompose_matrices()` in `src/model_reduction.jl`. So it gets converted twice into Hz, which might explain the difference in testing.

Frosco commented:
So, after a bit of back and forth the model expressed the following theory about the failing simple supported beam test:

The boundary conditions
```julia
bc = BridgeBC([
            [1, "trans"],
            [n_node, "y"]
        ])
```
represent a "clamped-simply supported beam", because for node 1, both x and y translations are constrained (clamped end) and for the n_node, only the y translation is constrained (simple support).

This means the analytical formula for the first natural frequency should be changed to f₁ = (3.9266²/2π) × √(EI/(ρAL⁴)).
If we do that, everything aligns quite nicely and the tests pass.

What do you think about that?

jgrashorn commented:
I have an issue with how the LLM isn't able to understand what a simply supported beam is based on the given boundary conditions, but it is nice to know where the limitations are :D
I have no issue with the test if it works, though.

Frosco committed:
Changed test expectation to being a clamped-simply supported beam

Frosco committed:
Removed Hz conversion

It's already done in the code that is tested, specifically in decompose_matrices()

Frosco commented:
> I have an issue with how the LLM isn't able to understand what a simply supported beam is based on the given boundary conditions, but it is nice to know where the limitations are :D I have no issue with the test if it works, though.

Yeah, these things still make up stuff as they go :woman_shrugging: 
But this is also part of the exercise: If we can overcome the limitations if we have enough control loops in our process ;) 

Anyway, stuff should work now, ready to merge :)
